src/hooks/Guilds/guild/useProposalCalls.ts

```diff
@@ -34,27 +37,50 @@ const useProposalCalls = (guildId: string, proposalId: string) => {
   } = proposal || {};
 
   const totalOptionsNum = totalVotes?.length || 0;
+  const displayableOptionsNum = totalOptionsNum - 1;
+  const nonApprovalOrEmptyCalls = useMemo(
+    () => dataArray?.filter(data => !isApprovalData(data) && !isZeroHash(data)),
+    [dataArray]
+  );
+
   const callsPerOption = totalOptionsNum
-    ? toArray?.length / totalOptionsNum
+    ? nonApprovalOrEmptyCalls.length / displayableOptionsNum
     : 0;
   const optionLabels = metadata?.voteOptions;
 
   const calls: Call[] = useMemo(() => {
-    return toArray?.map((to, index) => ({
+    const buildCall = idx => ({

```

Comment:
❓ What do you think about adding types to this function?

Reply:
Yep, done!